### PR TITLE
Revamp docs

### PR DESCRIPTION
**Pull request by lucperkins**
This feels like maybe a bit overkill, but I wanted to really hammer home a template that we can use for the other Actions (or future Actions). Please feel 100% free to push back on anything here.

The rendered README is here:

https://github.com/bufbuild/buf-breaking-action/tree/lperkins/docs-combover

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,11 +51,25 @@ Parameter | Description | Required | Default
...
+need to be properly formed Inputs, that is, `buf` needs to be able to [build][buf-build] both into
+an [Image]. You can verify this locally using the [`buf build`][buf-build] command on both Inputs.
+Some examples:
+
+```sh
+# Build the `main` branch
+buf build .git#main
+
+# Build a feature branch
+buf build .git#v0.1.0
```
Comment (amckinney) [inline README.md:68]: We can include an example with `branch` and ref`.

```suggestion
buf build .git#branch=main

# Build the v0.1.0 feature tag
buf build .git#ref=v0.1.0
```

Comment (amckinney) [reply README.md:68]: Feel free to update the comment language - the `branch` and `ref` bits are the important bits.

**Review by amckinney — APPROVED** [review]
Looks great overall, just a small comment on the Inputs example.

File: README.md (hunk 1 above)
Comment (lucperkins) [reply README.md:68]: Good call

**Pull request merged**